src/renderer/features/assets/AssetsPortfolio/model/portfolio-model.ts

```diff
@@ -0,0 +1,174 @@
+import { createEffect, createEvent, createStore, sample } from 'effector';
+import { once } from 'patronum';
+
+import { Account, AccountId, Balance, Chain, ChainId, TokenAsset, TokenBalance, Wallet } from '@shared/core';
+import { networkModel, networkUtils } from '@entities/network';
+import { accountUtils, walletModel, walletUtils } from '@entities/wallet';
+import { AssetsListView } from '@entities/asset';
+import { balanceModel, balanceUtils } from '@entities/balance';
+import { sumTokenBalances, tokensService } from '../lib/tokensService';
+
+const setActiveView = createEvent<AssetsListView>();
+const setAccounts = createEvent<Account[]>();
+
+const $activeView = createStore<AssetsListView | null>(null);
+const $accounts = createStore<Account[]>([]);
+const $tokens = createStore<TokenAsset[]>([]);
+const $activeTokens = createStore<TokenAsset[]>([]);
+
+const updateTokensFx = createEffect(
+  ({ activeWallet, chains }: { activeWallet?: Wallet; chains: Record<ChainId, Chain> }): TokenAsset[] => {
+    const tokens = tokensService.getTokensData();
+
+    return tokens.reduce((acc, token) => {
+      const filteredChains = token.chains.filter((chain) => {
+        return activeWallet?.accounts.some((account) => {
+          return (
+            accountUtils.isNonBaseVaultAccount(account, activeWallet) &&
+            accountUtils.isChainAndCryptoMatch(account, chains[chain.chainId])
+          );
+        });
+      });
+
+      if (filteredChains.length > 0) {
+        acc.push({ ...token, chains: filteredChains });
+      }
+
+      return acc;
+    }, [] as TokenAsset[]);
+  },
+);
+
+const populateTokensBalanceFx = createEffect(
+  ({
+    activeTokens,
+    balances,
+    accounts,
+  }: {
+    activeTokens: TokenAsset[];
+    balances: any;
```

Comment:
Is there a reason why this can't be typed `Balance[]`? I tried locally and seemed fine.